File: client/src/app/shared/SubjectGrade.ts
import { IStudent } from "./Student";
import { ISubject } from "./Subject";

export class ISubjectGrade {
    subjectGradeId: number;
    studentId: number;
    student: IStudent;
    grade: string;
    gradePoint: number;
    subjectId: number;
    subject: ISubject
}

File: Data/Entities/SubjectGrade.cs
namespace Magnifinance.Data.Entities
{
    public class SubjectGrade
    {
        public int SubjectGradeId { get; set; }
        public int StudentId { get; set; }
        public Student Student { get; set; }

        //public int GradeId { get; set; }
        //public Grade Grade { get; set; }

        public string Grade { get; set; }
        public double GradePoint { get; set; }

        public int SubjectId { get; set; }
        public Subject Subject { get; set; }
    }
}
